lib/src/components/NxProgressBar/NxProgressBar.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { faExclamationCircle, faCheckCircle } from '@fortawesome/free-solid-svg-icons';
+import classnames from 'classnames';
+import React, { forwardRef } from 'react';
+
+import { useUniqueId } from '../../util/idUtil';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { Props, propTypes } from './types';
+
+import './NxProgressBar.scss';
+
+const NxProgressBar = forwardRef<HTMLProgressElement, Props>(
+    function NxProgressBar(props, ref) {
+      const {
+        hasError,
+        inlineCounter = false,
+        label,
+        labelError,
+        labelSuccess,
+        max: maxProp,
+        showCounter = true,
+        value,
+        variant
+      } = props;
+
+      const max = maxProp || 100;
+      const percentage = Math.round(value / max * 100);
+
+      const progressId = useUniqueId('nx-progress-bar__progress');
+
+      const renderCounter = () => {
+        if (!showCounter || variant === 'inline') {
+          return null;
+        }
+
+        return (
+          <div className="nx-progress-bar__counter">
+            <span className="nx-counter">{hasError ? 0 : percentage}%</span>
+          </div>
+        );
+      };
+
+      const renderLabel = () => {
+        if (variant === 'small') {
+          return null;
+        }
+
+        if (hasError) {
+          return labelError ? (
+            <>
+              <NxFontAwesomeIcon icon={faExclamationCircle} />
+              <span className="nx-progress-bar__label-text">{labelError}</span>
+            </>
+          ) : null;
+        }
+
+        if (percentage === 100 && (labelSuccess || label)) {
+          return (
+            <>
+              <NxFontAwesomeIcon icon={faCheckCircle} />
+              <span className="nx-progress-bar__label-text">{labelSuccess || label}</span>
+            </>
+          );
+        }
+
+        return <span className="nx-progress-bar__label-text">label</span>;
+      };
+
+      const renderCounterAndLabel = () => {
+        if (variant === 'inline') {
+          return null;
+        }
+
+        return (
+          <label htmlFor={progressId} className="nx-progress-bar__label">
```

Comment:
Could the `<label>` be the wrapping element rather than a sibling? Then you wouldn't need to deal with the id

Reply:
> Could the `<label>` be the wrapping element rather than a sibling? Then you wouldn't need to deal with the id

Made that change here: d59382ac8739c3784a9a9d2da10103eab3e3cc9b